server/text_generation_server/models/seq2seq_lm.py

```diff
@@ -574,23 +650,29 @@ def __init__(
         )
         tokenizer.bos_token_id = model.config.decoder_start_token_id
 
-        super(Seq2SeqLM, self).__init__(
+        self = cls.__new__(
+            cls,
+        )
+        super().__init__(
+            self,
             model_id=model_id,
             model=model,
             tokenizer=tokenizer,
             requires_padding=True,
             dtype=dtype,
             device=device,
         )
+        return self
 
     @property
     def batch_type(self) -> Type[Seq2SeqLMBatch]:
         return Seq2SeqLMBatch
 
-    def decode(self, decoder_ids: List[int]) -> str:
-        return self.tokenizer.decode(
-            decoder_ids, skip_special_tokens=True, clean_up_tokenization_spaces=False
-        )
+    # Not used anymore
+    # def decode(self, decoder_ids: List[int]) -> str:
+    #     return self.tokenizer.decode(
+    #         decoder_ids, skip_special_tokens=True, clean_up_tokenization_spaces=False
+    #     )
```

Comment:
```suggestion
```

Reply:
Isn't thtat surprising that it's not used anymore ?

Shouldn't we have those flags be used somewhere for those models ? Do we have test that cover the raison d'être of this code ?

Reply:
I think this code was to allow models to specify the value of `skip_special_tokens`. For example for santacoder, you needed the fill in the middle special tokens to correctly display the outputs.